src/words/environment/WordsObject.java

```diff
@@ -13,13 +15,35 @@
 	private Position cell;
 	private String currentMessage;
 	private Action lastAction;
-
+	private HashMap<WordsObject, ArrayList<Property>> referers;
+	
+	public void updateReferers(WordsObject referringObject, Property referringProperty) {
+		ArrayList<Property> referringProperties;
+		if ((referringProperties = referers.get(referringObject)) == null) {
```

Comment:
This is so C-like, it confused me for a bit, but I get what you're doing now.  :-)